src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -423,5 +423,13 @@ public static SyntaxToken WithoutTrivia(this SyntaxToken token)
         {
             return node.WithTrailingTrivia((IEnumerable<SyntaxTrivia>)trivia);
         }
+
+        /// <summary>
+        /// Attaches the node to a SyntaxTree that the same options as <paramref name="oldTree"/>
+        /// </summary>
+        internal static SyntaxNode InTreeWithOptionsFrom(this SyntaxNode node, SyntaxTree oldTree)
```

Comment:
>InTreeWithOptionsFrom [](start = 35, length = 21)

It feels like the name of this method isn't quite clear. Consider renaming. How does "AsRootOfNewTreeWithOptions" sound?

Reply:
Sounds good. I'll change